gammapy/background/tests/test_models.py

```diff
@@ -132,6 +137,106 @@ def make_test_array(empty=True):
+    assert_allclose(pie_fraction, 0)
+
+    # I have to use an other object excluded_region because the previous one was
+    # already sorted in the compute_pie_fraction
+    excluded_sources2 = make_excluded_sources()
+    source_closest = SkyCoord(excluded_sources2["RA"][1], excluded_sources2["DEC"][1], unit="deg")
+    separation = pointing_position.separation(source_closest).value
+    pie_fraction = compute_pie_fraction(excluded_sources, pointing_position, Angle(5, "deg"))
+    pie_fraction_expected = (2 * np.arctan(excluded_sources2["Radius"][1] / separation) / (2 * np.pi))
+    assert_allclose(pie_fraction, pie_fraction_expected)
+
+
+def test_select_events_outside_pie():
+    """
+    Create an empty image centered on the pointing position and all the radec position of the pixels will
+    define one event. Thus we create a false EventList with these pixels. We apply the select_events_outside_pie()
+    and we fill the image only with the events (pixels) outside the pie. We assert that outside the pie the image is
+    fill with 1 and inside with 0.
+    """
+    excluded_sources = make_excluded_sources()
+    excluded_sources2 = make_excluded_sources()
+    pointing_position = SkyCoord(0.5, 0.5, unit='deg')
+    # Define an empty image centered on the pointing position
+    image = make_empty_image(nxpix=1000, nypix=1000, binsz=0.01, xref=pointing_position.ra.deg,
+                             yref=pointing_position.dec.deg,
+                             proj='TAN', coordsys='CEL')
+    ra, dec = coordinates(image)
+    events = EventList()
+    # Faked EventList with the radec of all the pixel in the empty image
+    events["RA"] = ra.flat
+    events["DEC"] = dec.flat
+
+    # Test that if the sources are out of the fov, it gives the index for all the events since no event will be removed
+    idx = select_events_outside_pie(excluded_sources, events, pointing_position, Angle(0.3, "deg"))
+    assert_allclose(np.arange(len(events)), idx)
+
+    # Test if after calling the select_events_outside_pie, the image is 0 inside the pie and 1 outside the pie
+    idx = select_events_outside_pie(excluded_sources, events, pointing_position, Angle(5, "deg"))
+    events_outside = events[idx]
+    image = bin_events_in_image(events_outside, image)
+    # image.writeto("test_pie.fits", clobber=True)
+    source_closest = SkyCoord(excluded_sources2["RA"][1], excluded_sources2["DEC"][1], unit="deg")
+    separation = pointing_position.separation(source_closest)
+    phi_source = pointing_position.position_angle(source_closest)
+    radius = Angle(excluded_sources2["Radius"])[1]
+    phi_min = phi_source - np.arctan(radius / separation)
+    phi_max = phi_source + np.arctan(radius / separation)
+    skycoord_pix = SkyCoord(ra, dec, unit='deg')
+    phi = pointing_position.position_angle(skycoord_pix)
+    idx_image_out_pie = np.where((phi > phi_max) | (phi < phi_min))
```

Comment:
You're re-implementing the algorithm in the test.
That's problematic ... it's a lot of complex extra code to read / maintain.
And if there's a bug in the actual function, there's a good chance you'd make the same error when writing the test.

I do think it's good that you've done this check and manually verified that you get a pie image with this function!

But for the test here, maybe just put a few positions (a handful or less) where you know they are inside or outside the pie and assert that you get the right answer?

I know we talked about the image as a test before.
But do you agree that re-implementing the algorithm in the test is bad?